traffic-policy/actions/oidc/config.mdx

```diff
@@ -0,0 +1,110 @@
+	>
+		<p>
+			A map of additional URL parameters to apply to the authorization endpoint
+			URL.
+		</p>
+	</ConfigItem>
+	<ConfigItem title="max_session_duration" type="duration" required={false}>
+		<p>Defines the maximum lifetime of a session regardless of activity.</p>
+	</ConfigItem>
+	<ConfigItem title="idle_session_duration" type="duration" required={false}>
+		<p>
+			Defines the period of inactivity after which a user's session is
+			automatically ended, requiring re-authentication.
+		</p>
+	</ConfigItem>
+	<ConfigItem
+		title="userinfo_refresh_interval"
+		type="duration"
+		required={false}
+	>
+		<p>
+			How often should ngrok refresh data about the authenticated user from the
+			identity provider.
+		</p>
+	</ConfigItem>
+	<ConfigItem title="allow_cors_preflight" type="boolean" required={false}>
+		<p>
+			Allow CORES preflight requests to bypass authentication checks. Enable if
+			the endpoint needs to be accessible via CORES.
+		</p>
+	</ConfigItem>
+	<ConfigItem title="inject_headers" type="boolean" required={false}>
+		<p>If false, we don't inject any headers.</p>
+	</ConfigItem>
+</Config>
+
+### Special Paths
+
+Upstream applications behind endpoints with this module enabled do not receive
+any requests to paths beginning with `/auth/`. Your application may redirect
+clients to the following paths to invoke different behaviors.
+
+| Path            | Description                                                                                                                                                                                                                                                                                                                     |
+| --------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `/ngrok/login`  | Redirect users to this path to explicitly begin an authentication flow. After authentication, users will be redirected to `/`. If the IdP supports it, ngrok will attempt to instruct the IdP to force re-authentication which will force users to re-enter their credentials with the IdP even if they were already logged in. |
+| `/ngrok/logout` | Logs the user out by clearing their session cookie. Redirect users to this path to log them out.                                                                                                                                                                                                                                |
+
+### Events
+
+When this module is enabled, it populates the following fields in the
+[http_request_complete.v0](/obs/reference/#http-request-complete) event:
+
+| Fields                |
+| --------------------- |
+| `oauth.app_client_id` |
```

Comment:
is this oidc.app_client_id?

Reply:
From my investigation it looks like they're both under `oauth` for the event system.